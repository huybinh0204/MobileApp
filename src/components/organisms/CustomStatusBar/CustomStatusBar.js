import PropTypes from 'prop-types';
import React from 'react';
import { Platform, StatusBar } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { StatusBarContainer } from './CustomStatusBar.styles';

const CustomStatusBar = ({ backgroundColor, ...props }) => {
  return Platform.OS === 'ios' ? (
    <StatusBarContainer backgroundColor={backgroundColor}>
      <SafeAreaView edges={['top']}>
        <StatusBar backgroundColor={backgroundColor} {...props} />
      </SafeAreaView>
    </StatusBarContainer>
  ) : (
    <StatusBar backgroundColor={backgroundColor} {...props} />
  );
};

CustomStatusBar.propTypes = {
  backgroundColor: PropTypes.string.isRequired,
};

export default CustomStatusBar;
