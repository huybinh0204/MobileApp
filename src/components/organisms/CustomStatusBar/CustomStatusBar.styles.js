import styled from '@emotion/native';
import { StatusBar } from 'react-native';

export const StatusBarContainer = styled.View(({ backgroundColor }) => ({
  backgroundColor,
  height: StatusBar.currentHeight,
}));
