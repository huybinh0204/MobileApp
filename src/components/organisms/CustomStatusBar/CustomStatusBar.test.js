import { render } from '@testing-library/react-native';
import React from 'react';
import Theme from '_utilities/Theme';
import CustomStatusBar from './CustomStatusBar';

describe('CustomStatusBar', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(
      <Theme>
        <CustomStatusBar backgroundColor="#ECF6EA" />
      </Theme>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
