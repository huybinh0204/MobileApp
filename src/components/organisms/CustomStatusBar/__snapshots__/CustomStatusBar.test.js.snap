// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomStatusBar should match the snapshot 1`] = `
<View
  backgroundColor="#ECF6EA"
  style={
    Array [
      Object {
        "backgroundColor": "#ECF6EA",
        "height": null,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaView
    edges={
      Array [
        "top",
      ]
    }
  />
</View>
`;
